Sources/SwiftDocC/Semantics/MarkupReferenceResolver.swift

```diff
@@ -80,6 +80,13 @@ struct MarkupReferenceResolver: MarkupRewriter {
             return resolved
             
         case .failure(let unresolved, let error):
+            if let rangeLowerBoundSource = range?.lowerBound.source,
+               let rangeUpperBoundSource = range?.upperBound.source,
+               let source,
```

Comment:
This `if let` shorthand was added in Swift 5.7 ([SE-0345][1]) but DocC still uses Swift 5.6. 

[1]: https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md

```suggestion
               let source = source,
```

Reply:
The Swift 5.7 bump for swift-markdown is discussed in a [workgroup meeting](https://forums.swift.org/t/documentation-workgroup-meeting-july-17th-2023/66170/3). 

Although the inspiration is from swift-markdown, and [the PR](https://github.com/apple/swift-markdown/pull/135) has not been merged for some CR reasons. But according to the SDWG meeting's notes, I think it's OK to bump the version in this PR.

> Kyle: Is it okay to bump the Swift tools version for Swift-Markdown?
> Franklin: Yes, we want to maintain compatibility for folks that haven't had time to update yet, but Swift 5.7 was released a year ago. It's okay for DocC to be using more recent versions since it's primary a toolchain product. 

If you have concerns about this and feel that this bump is not necessary for this PR, we can also update the commit instead.

Reply:
If we want to cherry-pick this fix into the 5.10 release we probably won't want the Swift bump but it's just one line that'd need to be modified in the cherry-pick PR so it doesn't really matter.

Reply:
Got it. I'll open a new PR to bump the version instead when needed.